include/seqan3/search/fm_index/bi_fm_index_cursor.hpp

```diff
@@ -437,6 +437,12 @@ class bi_fm_index_cursor
         return false;
     }
 
+    //!\overload
+    bool extend_right(char const * c) noexcept
```

Comment:
this should work for all char types:

```cpp
    template <typename char_type>
    //!\cond
        requires detail::is_char_adaptation_v<char_type>
    //!\endcond
    bool extend_right(char_type const * cstring) noexcept
    {
        return extend_right(std::string_view{cstring});
    }
```
same for all the other overloads

Reply:
If we go this route:
1) discuss the name and the location of `is_char_adaptation_v` in the core-team (this concept is private for our adaptation of chars for alphabets)
2) ```c++
    return extend_right(std::string_view{cstring});
    // -> should be
    return extend_right(std::basic_string_view<char_type>{cstring});
    ```

Reply:
> 1. discuss the name and the location of is_char_adaptation_v in the core-team (this concept is private for our adaptation of chars for alphabets)

This can be part of another PR

> 2. ...

Yes! Thanks

Reply:
2. I have rewritten the function as mentioned, but it does not find ```is_char_adaptation_v```.
```bash
In file included from ...seqan3/include/seqan3/search/fm_index/fm_index.hpp:26,
                 ...seqan3/include/seqan3/search/fm_index/bi_fm_index.hpp:20,
                 ...seqan3/include/seqan3/search/fm_index/all.hpp:48,
                 ...seqan3/include/seqan3/search/algorithm/search.hpp:19,
                 ...seqan3/include/seqan3/search/algorithm/all.hpp:67,
                 ...seqan3/test/unit/search/search_test.cpp:11:
...seqan3/include/seqan3/search/fm_index/fm_index_cursor.hpp:288:26: error: 'is_char_adaptation_v' is not a member of 'seqan3::detail'
  288 |         requires detail::is_char_adaptation_v<char_type>
      |                          ^~~~~~~~~~~~~~~~~~~~
...seqan3/include/seqan3/search/fm_index/fm_index_cursor.hpp:288:56: error: expected primary-expression before '>' token
  288 |         requires detail::is_char_adaptation_v<char_type>
      |                                                        ^
...seqan3/include/seqan3/search/fm_index/fm_index_cursor.hpp:290:5: error: expected primary-expression before 'bool'
  290 |     bool extend_right(char_type const * cstring) noexcept
      |     ^~~~
```

Reply:
use: `#include <seqan3/alphabet/adaptation/char.hpp>`